### DISCUSSION TITLE
Playground for Service Agreement

### DISCUSSION THREAD

**Discussion by mzxyz** [subquery/subql#1071] [Frontier TestNet - Bug Reporting]
We now have 2 running indexers which are enable authorisation for the query services, that means only consumers have valid service agreements with the indexer can consume the query service with auth token.

These 2 indexers:

- `0x597b16Be71ACD6973F896dE7fE0EC35F6bf78Dd7`
- `0x4baDAF26fd2Ad4039Da8Ff2434A53dA0860c6e19`

**Comment by mzxyz** [comment]
## Instruction for using playground with valid service agreement

1. Go to either Polkadot or Kusuma project page, search one of the above indexers and send the transaction to purchase an offer.
<img width="1543" alt="Screen Shot 2022-05-26 at 9 37 33 PM" src="https://user-images.githubusercontent.com/8177474/170462607-a56d1bda-3872-4065-b4b7-7be903e10f75.png">

2. Once transaction complete, go to my service agreements page, the agreement will display after around 20s.
<img width="1542" alt="Screen Shot 2022-05-26 at 9 38 16 PM" src="https://user-images.githubusercontent.com/8177474/170462828-18725fac-845e-4122-ab21-3360972238ca.png">

3. Press the `playground`  button and jump to playground page
<img width="1544" alt="Screen Shot 2022-05-26 at 9 38 40 PM" src="https://user-images.githubusercontent.com/8177474/170462954-cbe9b96f-7e05-422b-beb3-3ba241228419.png">

4. Press `Request Token` to sign a message with MetaMask and request the auth token
<img width="1549" alt="Screen Shot 2022-05-26 at 9 38 58 PM" src="https://user-images.githubusercontent.com/8177474/170463155-b6d75e16-6476-4584-bbe0-ae1e9b56dd94.png">

5. Try to play with the default query and build you own query
<img width="1523" alt="Screen Shot 2022-05-26 at 9 40 09 PM" src="https://user-images.githubusercontent.com/8177474/170463401-5bb941a0-8811-4f52-a76e-b5fa1c632f35.png">

**Comment by vvayffaring** [comment]
`0x7C...3212`
`0x6B...c91b`
